zerver/tests/test_markdown.py

```diff
@@ -1500,12 +1500,43 @@ def test_multiple_matching_realm_patterns(self) -> None:
             converted.rendered_content,
             '<p>We should fix <a href="https://trac.example.com/ticket/ABC-123">ABC-123</a> or <a href="https://trac.example.com/ticket/16">trac ABC-123</a> today.</p>',
         )
-        # But both the links should be generated in topics.
+        # Only the older linkifier should be used in the topic, because the two patterns overlap.
         self.assertEqual(
             converted_topic,
             [
                 {"url": "https://trac.example.com/ticket/ABC-123", "text": "ABC-123"},
-                {"url": "https://other-trac.example.com/ticket/ABC-123", "text": "ABC-123"},
+            ],
+        )
+
+        url_format_string = r"https://yet-another-trac.example.com/ticket/%(id)s"
+        linkifier_3 = RealmFilter(
+            realm=realm,
+            pattern=r"(?P<id>[A-Z][A-Z0-9]+)",
+            url_format_string=url_format_string,
+        )
+        linkifier_3.save()
+        self.assertEqual(
+            str(linkifier_3),
+            r"<RealmFilter(zulip): (?P<id>[A-Z][A-Z0-9]+)"
+            " https://yet-another-trac.example.com/ticket/%(id)s>",
+        )
+
+        msg = Message(sender=self.example_user("othello"))
+        msg.set_topic_name("ASD-123 ABC-123 QWE")
+
+        flush_per_request_caches()
+
+        content = (
+            "We should fix ASD-123 or [trac ASD-123](https://trac.example.com/ticket/16) today."
+        )
+        converted = markdown_convert(content, message_realm=realm, message=msg)
```

Comment:
This test seems to not inspect `converted`; is that intended?

Reply:
Updated